jupyter/utility/s3_buckets_analysis.py

```diff
@@ -0,0 +1,331 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+# %pip install boto3
+import pip
+import itertools
+import pandas as pd
+import os
+
+class StopExecution(Exception):
+    def _render_traceback_(self):
+        pass
+
+
+def import_or_install(package):
+    try:
+        __import__(package)
+        print(f'{package} is already installed')
+    except ImportError:
+        print(f'Installing {package}')
+        pip.main(['install', package])
```

Comment:
if one runs the notebook from the scratch one cells after another - will it work?

Reply:
Yes it will Though the cells are more or less independent of each other. They contain comments about what they do.